intel/custom_resource.py

```diff
@@ -0,0 +1,213 @@
+                "shortNames": self.short_names
+            }
+        }
+        body = {
+            "metadata": {"name": ".".join([self.plural_name, self.group])},
+            "spec": spec
+        }
+
+        try:
+            header_params = {
+                'Content-Type': "application/json",
+                'Accept': "application/json"
+            }
+
+            auth_settings = ['BearerToken']
+
+            resource_path = \
+                '/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions'
+            self.api.api_client.call_api(
+                resource_path,
+                'POST',
+                header_params,
+                body=body,
+                auth_settings=auth_settings)
+
+        except K8sApiException as e:
+            if json.loads(e.body)["reason"] != "AlreadyExists":
+                raise e
+
+        # Wait until resource type is ready.
+        while not self.exists():
+            logging.info(
+                "waiting 1 second until third party resource is ready")
+            time.sleep(1)
+
+    def exists(self, namespace="default"):
+        header_params = {
+            'Content-Type': "application/json",
+            'Accept': "application/json"
+        }
+
+        auth_settings = ['BearerToken']
+
+        # /apis/stable.example.com/v1/namespaces/*/crontabs
+        resource_path = "/".join([
+            "/apis",
+            self.group,
+            "v1",
+            "namespaces", namespace,
+            self.plural_name
+        ])
+
+        try:
+            self.api.api_client.call_api(
+                resource_path,
+                'GET',
+                header_params,
+                auth_settings=auth_settings)
+
+        except K8sApiException as e:
+            if json.loads(e.body)["reason"] == "NotFound":
+                return False
+            raise e
+
+        return True
+
+    def create(self, name, namespace="default"):
+        self.save()
+        return CustomResourceDefinition(self.api, self, namespace, name)
+
+
+class CustomResourceDefinition:
+    def __init__(self, api, resource_type, namespace, name):
+        assert (api is not None)
+        assert (resource_type is not None)
+        assert (name is not None)
+        assert (namespace is not None)
+
+        self.api = api
+        self.resource_type = resource_type
+        self.name = util.ldh_convert_check(name)
+        self.namespace = namespace
+
+        self.body = {
+            "apiVersion": "/".join([
+                self.resource_type.group,
+                self.resource_type.type_version
+            ]),
+            "kind": self.resource_type.kind_name,
+            "metadata": {"name": self.name},
+            "spec": {}
+        }
+
+        self.header_params = {
+            'Content-Type': "application/json",
+            'Accept': "application/json"
+        }
+
+        self.auth_settings = ['BearerToken']
+
+        self.resource_path = "/".join([
```

Comment:
Why do you bind all this variables to `self` here and in the above class create them in the `save` method scope (local variables)?
Please be consistent.